### PR TITLE
[deploy-agent] Return type hints in staging & types folders

### PR DESCRIPTION
**Pull request by gzpcho**
Improve readability for low risk:

> The Python runtime does not enforce function and variable type annotations.

### CODE REVIEW

File: deploy-agent/deployd/staging/transformer.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def _translate(self, from_path, to_path):
             log.error('Fail to translate script {}, stacktrace: {}'.format(from_path,
                                                                            traceback.format_exc()))
 
-    def transform_scripts(self, script_dir, template_dirname, script_dirname):
+    def transform_scripts(self, script_dir, template_dirname, script_dirname) -> list:
```
Comment (ntascii) [inline deploy-agent/deployd/staging/transformer.py:76]: probably we should be consistent to use either list or List from typing. wdyt?

File: deploy-agent/deployd/types/deploy_goal.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def __init__(self, jsonValue=None):
             self.firstDeploy = jsonValue.get('firstDeploy')
             self.isDocker = jsonValue.get('isDocker')
 
-    def __key(self):
+    def __key(self) -> tuple:
```
Comment (ntascii) [inline deploy-agent/deployd/types/deploy_goal.py:54]: same with tuple and Tuple from typing

File: deploy-agent/deployd/staging/transformer.py (hunk 1 above)
Comment (gzpcho) [reply deploy-agent/deployd/staging/transformer.py]: good catch, thanks!

File: deploy-agent/deployd/types/deploy_goal.py (hunk 2 above)
Comment (gzpcho) [reply deploy-agent/deployd/types/deploy_goal.py]: good catch, thanks!

**Review by ntascii — APPROVED** [review]

**Pull request merged**